include/zephyr/bluetooth/testing_cfg.h

```diff
@@ -0,0 +1,39 @@
+/* Copyright (c) 2024 Nordic Semiconductor
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file Runtime configuration for Bluetooth testing purposes.
+ */
+
+#include <stdbool.h>
+
+#include <zephyr/sys/util_macro.h>
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_TESTING_CFG_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_TESTING_CFG_H_
+
+struct bt_testing_cfg {
+	bool l2cap_ecred: 1;
+};
+
+static const struct bt_testing_cfg bt_testing_cfg_all_enabled = {
+	.l2cap_ecred = true,
+};
+
+extern struct bt_testing_cfg _bt_testing_cfg_runtime_val;
+
+static inline struct bt_testing_cfg bt_testing_cfg_get(void)
```

Comment:
As the struct grows, wouldn't it make more sense to return it by reference rather than by value from the start?

Reply:
For performance reasons? Since it's an inline function, there should be no difference in the generated code.

Reply:
That was my first though, however it's also more common (the consistency argument again) to pass structs by reference rather than by value. I.e. this would be make the code look "less odd", at least to me :)

Reply:
I agree with you then. I will choose pointers over pass-by-value for consistency. But when I was about to do this, though it might be better to just make it a simple variable without any getter or setter. It's not a public API after all. Please have a look at the new push and tell we what you think!